exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetReaderUtility.java

```diff
@@ -184,17 +201,15 @@ public static DateCorruptionStatus detectCorruptDates(ParquetMetadata footer,
           if (parsedCreatedByVersion.hasSemanticVersion()) {
             SemanticVersion semVer = parsedCreatedByVersion.getSemanticVersion();
             String pre = semVer.pre + "";
-            if (semVer != null && semVer.major == 1 && semVer.minor == 8 && semVer.patch == 1 && pre.contains("drill")) {
```

Comment:
I don't know if you are guarding against a null value for semVer at a higher level, but I'm pretty sure the reason I added it is that older versions of parquet files can be lacking this field. I assume with all of the tests cases that were added we should be okay, but it might be better to keep it in defensively. If this field isn't set in the metadata, it shouldn't make it an invalid file, but if we get an NPE it will stop query processing.